src/config.ts

```diff
@@ -38,27 +37,11 @@ export function defineConfig(config: ViteUserConfig = {}): ViteUserConfig {
           singleFork: true,
         },
       },
-      forceRerunTriggers: [
-        ...(_config.test?.forceRerunTriggers
-          ? []
-          : [
-              // Vitest defaults
-              '**/package.json/**',
-              '**/{vitest,vite}.config.*/**',
-            ]),
-        // Re-run tests when Nitro is rebuilt
-        join(
-          nitro?.rootDir || '',
-          NITRO_OUTPUT_DIR,
-          nitro?.mode === 'production' ? 'server' : '.nitro/dev',
-          'index.mjs',
-        ),
-      ],
-      globalSetup: [
-        join(currentDir, 'setup.mjs'),
-      ],
-      // @ts-expect-error: Append Nitro config to access in global setup file
-      nitro,
+      forceRerunTriggers: nitro?.forceRerunTriggersOnSrcDir
+        ? [
+        `${(await loadNitroOptions()).srcDir}/**/*.ts`,
```

Comment:
I think it is safer to only rerun tests when the final dev/prod server bundle has been rebuilt. Don't you think?

```ts
join(
  nitro?.rootDir || '',
  NITRO_OUTPUT_DIR,
  nitro?.mode === 'production' ? 'server' : '.nitro/dev',
  'index.mjs',
)
```

Then, we could drop the `forceRerunTriggersOnSrcDir` option altogether.

Reply:
> only rerun tests when the final dev/prod server bundle

The problem with that solution is:

1. Tests starts
2. Server is built => `index.mjs` is changed (something is pushed to a queue for later)
3. Tests run
4. The queue is read and this trigger a server change => back to step 2

This behavior will loop over and over again.

Also, I think it's important to allow dev to remove this behavior (addition of forceReruntrigger) if needed.

Reply:
Good catch! Thanks for the explanation.